feat(Shiro): upgrade shiro version 1.7.1 -> 1.13.0

animator013 opened the pull request:
Hi there,

here is PR for the shiro upgrade to latest jdk8 version for Grails 3.

Issue #20 

Please have a look.

The tests were running version 3.3, and main version was 3.4, not sure if that was a bug, maybe this should copy the main version, I changed to 3.4 and I think you should change it to the main again.

animator013 committed:
feat(Shiro): upgrade shiro version 1.7.1 -> 1.13.0

pmcneil commented:
Looks good, did the tests run? Not sure what you mean by the tests were running 3.3, it should run the latest version.

pmcneil commented:
OK Haven't looked at the grail 3 plugin for a bit, just firing it up. Looks like there are some test config issues (mostly me getting the env right). Hopefully I'll get it up and running tomorrow. 

It looks like the test generation should be "compileOnly..." not "compile ..."

animator013 commented:
Hi there,

i ran the tests and all were successful. But I don't know if it is my environment specific (I use WSL2 inside Windows), well the tests of LDAP are not reliable because the LDAP server either shuts itself or it refuses connection or test freezes. It took a few tries so that tests were successful.

Many times i get stuck here:
```
:shiroTest > 6 tests completed
:shiroTest > Executing test org.apache.shiro.grails.LdapServerSpec
```

After complete WSL reset i got 100% test cases. 

![image](https://github.com/user-attachments/assets/d1efa735-3e9a-4c49-8600-74cd135420b1)

animator013 commented:
> Looks good, did the tests run? Not sure what you mean by the tests were running 3.3, it should run the latest version.

If you look into my commit the "mini projects" were using `compile "org.grails.plugins:grails-shiro:3.3` but the version of shiro is currently `3.4`. I updated it in the `build.gradle` to 3.4 and should be updated to next version you choose, I changed the version to 3.5 but versioning is up to you :)

pmcneil commented:
OK thanks I'll merge this pull request and also confirm.

animator013 commented:
Thanks, please let me now when the version is published. I can make another PR for docs update if you want.

pmcneil commented:
Docs update would be awesome.
